cardano-api/src/Cardano/Api/TxBody.hs

```diff
@@ -1618,11 +1622,14 @@ fromLedgerTxExtraKeyWitnesses sbe body =
     ShelleyBasedEraShelley -> TxExtraKeyWitnessesNone
     ShelleyBasedEraAllegra -> TxExtraKeyWitnessesNone
     ShelleyBasedEraMary    -> TxExtraKeyWitnessesNone
-    ShelleyBasedEraAlonzo  -> TxExtraKeyWitnesses
+    ShelleyBasedEraAlonzo
+      | null keyhashes     -> TxExtraKeyWitnessesNone
```

Comment:
Can we use `Set.null` here just to be explicit. The `Foldable` null overloading is a bit sketchy when it comes to the asymptotic performance. Yes, the set instance overrides it, but it's easy to make a representation change and not notice you've suddenly changed the asymptotic performance.

Reply:
Do you know containers with `null` slower than O(1)?